### DISCUSSION TITLE
Too Many Options for Saving

### DISCUSSION THREAD

**Discussion by robinxbaker** [audacity/audacity#6428] [General]
### Your idea

The issue I found was not a bug, but an issue regarding usability. The options for saving the audio are too lingually specific, meaning if someone is familiar with audio software, they will know what they’re looking at, but if they are beginners, which much of Audacity’s users are, it takes trial and error, and potential loss of the first few projects, to discover. Specifically, to save the audio for use as an audio file, one must go to file, where they will see “import audio”, “export audio”, “export audio as”, “save project”, and “save project as”.

### Problem to be solved

These are too many options for a beginner and can cause the user to select the wrong option, close Audacity under the pretense that they successfully saved the project and come to realize later that they lost everything they worked on because they selected “import audio” instead of “export audio as”. I want to simplify the process by having a “save audio (beginner)” and “save audio (advanced options)”, as having “beginner” and “advanced” options is a widely used practice currently that allows for a greater heuristic of users.

**Comment by petersampsonaudacity** [comment]
@robinxbaker 

> The options for saving the audio are too lingually specific

The Save/Open and Export/Import are deliberately differentiated and "specific".

With Save and Open you _**never**_  save or open an audio file - they **_only_** act on Audacity Projects (which in current Audacity happen to be SQLite databases).  They most certainly are **_not_** audio files, you cannot play an Audacity project in any player or application other than Audacity.

In contrast Export and Import do work on audio files, files that can be plated in audio players and other apps, that is their designed function.
a) and what Import does is to take the audio from the audio fuile and convert it inti its own internal project format
b) Export takes the Audacity project or selection and convets that to your specified audio format for storing as an audio file where you specify.

A long while ago some confusion was allowed to creep into Audacity whereby the Open command can be used on an audio file.  But this does not actully open the audio file, rather it **Opens** an Audacity project and the silently/hiddenly **Imports** the audio file into that project.  It also leads to the confusion of some users thinking thay are actually opening an audio file and working directly on it, which they are not.  _Many years ago I campaigned strongly to remove this irregularity, but I was over-ruled._

**Comment by LWinterberg** [comment]
The save vs export question is tricky, as half of the userbase only ever deals with audio files (so, import MP3, do some edit, export MP3) while the other half thinks in terms of projects. A "beginner" vs "expert" pattern likely isn't helpful here - both beginners and experts may save projects and both may export audio. 

> they lost everything they worked on because they selected “import audio” instead of “export audio as”.

This however is not an issue I ever have seen. If you import a file, it will do so to a new track, leaving you at a minimum confused what happened, but definitely leaving you certain that you didn't save properly. 

> Specifically, to save the audio for use as an audio file, one must go to file, where they will see “import audio”, “export audio”, “export audio as”, “save project”, and “save project as”.

This is not quite true. In the current version, they see this:

![image](https://github.com/audacity/audacity/assets/87814144/1356f051-cc42-4cc3-bb12-f10fa3e696eb)

The "Save Project" specific options and Import options are slightly hidden away into a submenu, and the Export Audio and Share Audio are top level entries you can see immediately.

**Comment by 8i99opiuhjk** [comment]
I know this thread is old but I just want to vent, as the IP is completely right. I am a beginner and Audacity's terrible 'save' functions has just caused me to lose 2 hours worth of editing. I have been going up to the save menu and just selecting 'save project' without realising that there is a sub menu? Why? Why not have those directly below the save project option? I have never seen a 'save' submenu on any software and it just inviting beginners to lose their work. So I have not been saving any of my work everytime I hit the save option. To make matters worse I found the 'export' function even more confusing. 

I go to the export panel, and then 'browse'. So I select my folder and there is another 'save' button. I assume that is going to save the file. Why not 'select'?? It's not 'saving' anything. So I wrongly assume when I return to the first dialogue it has saved. Because there is an 'export' button. Maybe that's so I can export under a different file type, I think. I would not expect to see the menu selection be "Export' - 'Export'.  This is muscle memory. Save means, save the file, not change the directory. Export is an option, not a save.

Then, if all that isn't bad enough, the final screw up. I think I've saved my file, and also exported it. Then I quit. Thinking that Audacity is one of those programs that constantly asks you to save changes, even when you've saved changes. But then, when it's all to late,  I realise, it's because I've been hitting the null 'Save' button without realising it's a sub-menu.

I know I'm stupid. I know it's all my fault. But I'm a beginner. These options do not make sense to me.

I sure as hell have learned now. Thanks.

**Reply by petersampsonaudacity** [reply]
> I know this thread is old but I just want to vent, as the IP is completely right. I am a beginner and Audacity's terrible 'save' functions has just caused me to lose 2 hours worth of editing. I have been going up to the save menu and just selecting 'save project' without realising that there is a sub menu? Why? Why not have those directly below the save project option? I have never seen a 'save' submenu on any software and it just inviting beginners to lose their work. So I have not been saving any of my work everytime I hit the save option.

@8i99opiuhjk 

If you are just saving a project as you work on it (and not bothering with Backup Project or Save As) then it's **_much_** easier just to use the **Ctrl+S** shortcut (**Cmd+S** on Mac).

**Reply by petersampsonaudacity** [reply]
> To make matters worse I found the 'export' function even more confusing.
> 
> I go to the export panel, and then 'browse'. So I select my folder and there is another 'save' button. I assume that is going to save the file. Why not 'select'?? It's not 'saving' anything. So I wrongly assume when I return to the first dialogue it has saved.

AFAIK that is because Audacity is using the Microsoft file/directory picker (rather than writing and managing it's own picker).

**Reply by petersampsonaudacity** [reply]
> Then, if all that isn't bad enough, the final screw up. I think I've saved my file, and also exported it. Then I quit. Thinking that Audacity is one of those programs that constantly asks you to save changes, even when you've saved changes.

Audacity certainly does that alright.  If you try to exit Audacity or close a project that has changes that have not been saved then you will get the following naggogram:
<img width="679" height="355" alt="image" src="https://github.com/user-attachments/assets/14b5afb5-9a39-46ca-8a03-a67037d7999d" />

_This naggogram has annoyed a fair few posters on the Forum over the years with them asking for a way to turn it off and not show again - but I don't think that's ever likely to happen.  Microsoft and Apple do likewise with their apps and most other apps that I know do likewise._
